lib/service/service.go

```diff
@@ -5683,3 +5687,10 @@ func copyAndConfigureTLS(config *tls.Config, log logrus.FieldLogger, accessPoint
 
 	return tlsConfig
 }
+
+func makeXForwardedForMiddleware(cfg *servicecfg.Config) utils.HTTPMiddleware {
+	if cfg.Proxy.TrustXForwardedFor {
+		return web.NewXForwardedForMiddleware
+	}
+	return nil
```

Comment:
It quite rare that `make...` function return nil without an error.  
Maybe we can return the no operation object to avoid nil panics error like:

```
	return func(next http.Handler) http.Handler {
		return next
	}
```